lib/msal-node/src/cache/NodeStorage.ts

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { ICacheStorage } from '@azure/msal-common';
+import { NodeCacheManager } from './NodeCacheManager';
+import { CacheOptions } from '../config/ClientConfiguration';
+
+/**
+ * This class implements the cache storage interface for MSAL through browser local or session storage.
+ * Cookies are only used if storeAuthStateInCookie is true, and are only used for
+ * parameters such as state and nonce, generally.
+ */
+export class NodeStorage implements ICacheStorage {
+    // Cache configuration, either set by user or default values.
+    private cacheConfig: CacheOptions;
+    private cachePath: string;
+    private nodeCacheManager: NodeCacheManager;
+
+    constructor(cacheConfig: CacheOptions) {
+        this.cacheConfig = cacheConfig;
+        this.cachePath = this.cacheConfig.cacheLocation!;
+        this.nodeCacheManager = new NodeCacheManager();
+    }
+
+    /**
+     * retrieve the file Path to read the cache from
+     */
+    getCachePath(): string {
+        return this.cachePath;
+    }
+
+    /**
+     * read JSON formatted cache from disk
+     */
+    getSerializedCache(): string {
```

Comment:
I think these should be async, since file IO should be async.

Reply:
Done.